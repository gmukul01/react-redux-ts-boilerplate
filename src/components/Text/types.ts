import { FontSizeTheme, FontWeightTheme } from '../../theme';
import { HTMLProps, WithThemeProp } from '../../utils/types';

interface Common {
    textColor?: string;
    uppercase?: boolean;
    lineThrough?: boolean;
}

export interface Props extends Common, HTMLProps<HTMLElement>, WithThemeProp {
    textSize?: keyof FontSizeTheme;
    textWeight?: keyof FontWeightTheme;
}

export interface StyledProps extends Common {
    textSize?: string;
    textWeight?: string;
}
