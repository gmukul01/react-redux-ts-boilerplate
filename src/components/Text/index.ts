import { TextStyled as Text } from './Text.styled';

export default Text;
