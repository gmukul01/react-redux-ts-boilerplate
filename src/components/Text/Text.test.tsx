import 'jest-styled-components';
import React from 'react';
import { create } from 'react-test-renderer';

import Text from './Text';

describe('Text Component', () => {
    it('should render span element by default', () => {
        const typography = create(<Text />);
        expect(typography.toJSON().type).toBe('span');
    });

    it('should render strong element for strong', () => {
        const typography = create(<Text textWeight="Strong" />).toJSON();
        expect(typography.type).toBe('strong');
    });

    it('should render with all the props', () => {
        const typography = create(<Text textSize="M2" textWeight="Light" textColor="pink" uppercase lineThrough />);
        expect(typography).toMatchSnapshot();
    });
});
