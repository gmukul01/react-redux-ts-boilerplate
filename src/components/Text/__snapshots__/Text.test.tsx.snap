// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Component should render with all the props 1`] = `
.c0 {
  font-size: 14px;
  font-weight: 200;
  color: pink;
  text-transform: uppercase;
  -webkit-text-decoration: line-through;
  text-decoration: line-through;
}

<span
  className="c0"
/>
`;
